src/main/java/com/groupseven/pdfproject/model/Selectable.java

```diff
@@ -5,12 +5,20 @@
 /**
  * @author Charles Witherspoon
  *
- *         \brief This interface represents a selectable canvas element \ref t18_1 "Task 18.1"
+ *         \brief This interface represents a selectable canvas element \ref t18_1 "Task 18.1" \ref t18_2 "Task 18.2"
  */
 public interface Selectable {
     void select();
 
     void unselect();
 
     Shape getSelection();
+
+    void setUri(String uri);
```

Comment:
I think it might be valuable to add another interface such as `Linkable` to represent objects that we can attach links to, rather than just adding the functionality into selectable.

I can see how linkable things must be selectable in some way, however I do wonder if everything that is selectable should be able to have a link attached to it.